stew/byteutils.nim

```diff
@@ -46,7 +46,9 @@ func hexToByteArrayImpl(hexStr: openArray[char], output: var openArray[byte], fr
   ## Allows specifying the byte range to process into the array
   var sIdx = skip0xPrefix(hexStr)
 
-  doAssert(fromIdx >= 0 and toIdx >= fromIdx and fromIdx < output.len and toIdx < output.len)
+  if fromIdx >= 0 and toIdx >= fromIdx and fromIdx < output.len and toIdx < output.len:
```

Comment:
we should only raise `ValueError` when `hexStr` is "invalid" - we should assert when `fromIdx`/`toIdx` are inconsistent / out-of-range with respect to `hexStr`

Reply:
also, a change like this should be accompanied by a test case

Reply:
Yep was just in the draft of this PR, sorry for notification